packages/grid/data-grid/src/DataGridProps.ts

```diff
@@ -14,11 +14,13 @@ export type DataGridProps = Omit<
   | 'disableMultipleColumnsFiltering'
   | 'disableMultipleColumnsSorting'
   | 'disableMultipleSelection'
+  | 'hideFooterRowCount'
   | 'options'
   | 'onRowsScrollEnd'
   | 'onViewportRowsChange'
   | 'scrollEndThreshold'
   | 'signature'
 > & {
   pagination?: true;
+  hideFooterRowCount?: false;
```

Comment:
Why setting `hideFooterRowCount` here instead of just removing it from the DataGrid interface ?

Reply:
I followed the pagination example. Should we remove it?

Reply:
https://github.com/mui-org/material-ui-x/pull/2176#issuecomment-887710864

I don't find it great to put prop that you can't do anything with in the interface.
So I removed all of them in #2176.

Except for `pagination` because it allows us to write

```tsx
<DataGrid pagination {...data} />
```

So the user cancopy/paste the code from a `DataGrid` demo into their `DataGridPro` and have the same pagination behavior.

But it is 100% equivalent to 

```tsx
<DataGrid {...data} />
```

But I still think that it's underlines a deeper problem, which is the different default value of `pagination` between `DataGrid` and `DataGridPro`.

I'll add a comment on the `pagination?: true` line :+1:

Reply:
```suggestion
```